sql/bookend.sql

```diff
@@ -26,12 +26,12 @@ LANGUAGE C IMMUTABLE;
 CREATE OR REPLACE FUNCTION _timescaledb_internal.bookend_serializefunc(internal)
 RETURNS bytea
 AS '$libdir/timescaledb', 'bookend_serializefunc'
-LANGUAGE C IMMUTABLE;
+LANGUAGE C IMMUTABLE STRICT;
```

Comment:
Are we sure this is safe? I mean it /seems/ safe but is there a way to test this in the parallel aggregate test suite (I guess this would require the aggregate to be null in those tests)? Or is this documented safe somewhere?

Reply:
The standard PostgreSQL tests mandate that all serialize/deserialize functions are STRICT, which seems right since it doesn't make sense to serialize NULL inputs.

I don't know why a test would make a difference since this function must previously have either (1) segfaulted or (2) returned NULL on NULL input. With STRICT set, the function is never invoked, returning NULL directly instead.

Reply:
Ahh I just reread the C code and there is indeed an !not null assert. You are right